kratos/elements/distance_calculation_flux_based_element.cpp

```diff
@@ -0,0 +1,502 @@
+// Public Operations
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim, TNumNodes >::CalculateLocalSystem(
+    MatrixType &rLeftHandSideMatrix,
+    VectorType &rRightHandSideVector,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // Resize and intialize output
+    if (rLeftHandSideMatrix.size1() != LocalSize)
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+
+    if (rRightHandSideVector.size() != LocalSize)
+        rRightHandSideVector.resize(LocalSize, false);
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+    const unsigned int step = rCurrentProcessInfo[FRACTIONAL_STEP];
+    
+    if(step == 1){ //solve a transcient diffusion problem
+        CalculatePotentialFlowSystem(rLeftHandSideMatrix, 
+                                     rRightHandSideVector,
+                                     rCurrentProcessInfo);
+    }
+    else if (step == 2){ // solve convection + source
+        CalculateDistanceSystem(rLeftHandSideMatrix, 
+                                     rRightHandSideVector,
+                                     rCurrentProcessInfo);
+    }
+    else{
+        KRATOS_ERROR << "FRACTIONAL_STEP must be 1 or 2" << std::endl;
+    }
+
+    KRATOS_CATCH("");
+}
+
+
+template <unsigned int TDim , unsigned int TNumNodes >
+void DistanceCalculationFluxBasedElement<TDim, TNumNodes >::CalculatePotentialFlowSystem(
+        MatrixType &rLeftHandSideMatrix,
+        VectorType &rRightHandSideVector,
+        const ProcessInfo &rCurrentProcessInfo)
+{
+    BoundedVector<double,TNumNodes> gauss_weights;
+    BoundedMatrix<double,TNumNodes,TNumNodes> N_container;
+    array_1d<BoundedMatrix<double, TNumNodes, TDim>, TNumNodes> DN_DX_container;
+
+    CalculateGaussPointsData( GetGeometry(), gauss_weights, N_container, DN_DX_container);
+
+    array_1d<double, NumNodes > nodal_values;
+    for(unsigned int i=0; i<NumNodes; i++){
+        nodal_values[i] = GetGeometry()[i].FastGetSolutionStepValue(DISTANCE);
+    }
+
+    //computing density so that Fourier number of the complete domain = 1
+    //this ensures that all regions notice the effect of the fixed nodes.
+    const double domain_length = rCurrentProcessInfo[CHARACTERISTIC_LENGTH];
+    const double density  = 1.0/ (domain_length*domain_length);
+
+    
+    //Looping Gauss Points
+    for (unsigned int igauss = 0; igauss<NumNodes; igauss++) {
+        array_1d<double, TNumNodes > N = row(N_container, igauss);
+        noalias(rLeftHandSideMatrix) += gauss_weights[igauss]*prod(DN_DX_container[igauss],trans(DN_DX_container[igauss]));
+        const double mass_factor = density*gauss_weights[igauss];
+        const double d_gauss = inner_prod(N,nodal_values);
+        const double initial_value = d_gauss > 0.0 ? domain_length*10.0 : -domain_length*10.0 ;
```

Comment:
Why multiplying by 10?

Reply:
you're right, it does not affect the final shape of the solution